hdf5-sys/build.rs

```diff
@@ -580,27 +596,68 @@ impl Config {
         vs.extend((0..=5).map(|v| Version::new(1, 10, v))); // 1.10.[0-5]
         for v in vs.into_iter().filter(|&v| version >= v) {
             println!("cargo:rustc-cfg=hdf5_{}_{}_{}", v.major, v.minor, v.micro);
+            println!("cargo:version_{}_{}_{}=1", v.major, v.minor, v.micro);
         }
         if self.header.have_stdbool_h {
             println!("cargo:rustc-cfg=h5_have_stdbool_h");
+            println!("cargo:have_stdbool=1");
         }
         if self.header.have_direct {
             println!("cargo:rustc-cfg=h5_have_direct");
+            println!("cargo:have_direct=1");
         }
         if self.header.have_parallel {
             println!("cargo:rustc-cfg=h5_have_parallel");
+            println!("cargo:have_parallel=1");
         }
         if self.header.have_threadsafe {
             println!("cargo:rustc-cfg=h5_have_threadsafe");
+            println!("cargo:have_threadsafe=1");
         }
     }
 }
 
 fn main() {
-    let mut searcher = LibrarySearcher::new_from_env();
-    searcher.try_locate_hdf5_library();
-    let config = searcher.finalize();
-    println!("{:#?}", config);
-    config.emit_link_flags();
+    if feature!("STATIC").is_ok() && !std::env::var_os("HDF5_DIR").is_some() {
+        get_build_and_emit();
+    } else {
+        let mut searcher = LibrarySearcher::new_from_env();
+        searcher.try_locate_hdf5_library();
+        let config = searcher.finalize();
+        println!("{:#?}", config);
+        config.emit_link_flags();
+        config.emit_cfg_flags();
+    }
+}
+
+fn get_build_and_emit() {
+    println!("cargo:rerun-if-changed=build.rs");
+
+    if feature!("ZLIB").is_ok() {
+        let zlib_lib = std::env::var("DEP_HDF5SRC_ZLIB").unwrap();
+        println!("cargo:zlib={}", &zlib_lib);
+        let zlib_lib_header = std::env::var("DEP_HDF5SRC_ZLIB").unwrap();
+        println!("cargo:zlib={}", &zlib_lib_header);
+        println!("cargo:rustc-link-lib=static={}", &zlib_lib);
+    }
+
+    if feature!("HL").is_ok() {
+        let hdf5_hl_lib = std::env::var("DEP_HDF5SRC_HL_LIBRARY").unwrap();
+        println!("cargo:rustc-link-lib=static={}", &hdf5_hl_lib);
+        println!("cargo:hl_library={}", &hdf5_hl_lib);
+    }
+
+    let hdf5 = std::env::var("DEP_HDF5SRC_ROOT").unwrap();
+    println!("cargo:root={}", &hdf5);
+    let hdf5_incdir = std::env::var("DEP_HDF5SRC_INCLUDE").unwrap();
+    println!("cargo:include={}", &hdf5_incdir);
+    let hdf5_lib = std::env::var("DEP_HDF5SRC_LIBRARY").unwrap();
+    println!("cargo:library={}", &hdf5_lib);
+
+    println!("cargo:rustc-link-search=native={}/lib", &hdf5);
```

Comment:
```suggestion
    println!("cargo:rustc-link-search=native={}/lib", &hdf5_root);
```

@magnusuMET Is using "/" ok here? E.g. on Windows

Reply:
This is supported, although I can't seem to find the documentation for this.